### PR TITLE
Assignment 2 ( Random Number , Taking Input , Function Expression , Arrow Function )

### PR DESCRIPTION
**Pull request by rishabh1164**

### CODE REVIEW

File: Assignment2.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+
+//Question 1 - Write a program to give a random output between 1 to 10 (AC: 2 simultaneous answers could not be same)
+let num = Math.floor(Math.random()*10 ) + 1;
```
Comment (ajay-nx) [inline Assignment2.js:3]: Please use const as you are not reassigning value to this variable. 
Please take care of naming convention & use meaningful variable names. 

ex - const randomNumber = Math.floor(Math.random() * 10) + 1;

File: Assignment2.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
+
+//Question 1 - Write a program to give a random output between 1 to 10 (AC: 2 simultaneous answers could not be same)
+let num = Math.floor(Math.random()*10 ) + 1; 
+// random fuction gives value between 0 and 1 (It is decimal value ), so I multiplied by 10 and add 1(because value must be in between 1 and 10 ) and   used Math.floor function to get the integer value 
+console.log(num);
+
+//Question 2 - Write a program to take input(age, name, phone no.) from user and display it
+//prompt function is used to get the input fro user in console window , this function might not give expected behaviour in vs code compiler
+let age = prompt("Enter your age");
```
Comment (ajay-nx) [inline Assignment2.js:9]: Please use const instead of let here.

File: Assignment2.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+console.log("age of " + name + "is" + age + "and phone no is " + phone);
+
+
+//Question 3 - Write a function expression that takes in another function as an argument
+//Here a is a function which takes num and a function as  arguments and printing them
+const a = function (num,func) {
+    console.log(func(num));
+}
+
+function numerical(value){
```
Comment (ajay-nx) [inline Assignment2.js:21]: Indentation is not proper - Please install extension for the better indentation of your code.

File: Assignment2.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    return num1+num2;
+}
+
+const ans = sum(2,13);//calling sum function and passing two numbers as parameters
+console.log(ans); // displaying ans on console window
+
+
+//Question 6- Write a function expression that takes in a number and returns true if it's even and false if it's odd.
+
+//cheker is a function expression which takes a value as an argument and returns true if its even and false otherwise
```
Comment (ajay-nx) [inline Assignment2.js:54]: if - else condition is un-necessary here - you can simply return the condition here - 

ex - const evenOddChecker = (value) => val % 2 === 0;

File: Assignment2.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+console.log(checker(10)); // calling checker function and printing ouput of checker function 
+
+
+
+
+//Question 7 - All of the above questions from 3-6 needs to be done with arrow functions also
+
+// Converting all the function expressions from question-3 to 6 into arrow function
+
+//Question 3 in Arrow Function
```
Comment (ajay-nx) [inline Assignment2.js:72]: Naming convention & indentation is not proper.

**Review by ajay-nx — CHANGES_REQUESTED** [review]

File: Assignment2.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
+
```
Comment (ajay-nx) [inline Assignment2.js:1]: Please read more about JS file naming convention & rename this file & other tasks files accordingly. 

ex - assignmentTwo.js
        assignment2.js
        assignment-2.js
        assignment_2.js
        
  Let's discuss on this tomorrow.

**Review by ajay-nx — CHANGES_REQUESTED** [review]

File: assignment2.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
+//Question 1 - Write a program to give a random output between 1 to 10 (AC: 2 simultaneous answers could not be same)
+
+const randomNumber = Math.floor(Math.random() * 10) + 1;
```
Comment (ajay-nx) [inline assignment2.js:3]: AC - Simultaneous answers could not be same is not achieved.

**Review by ajay-nx — CHANGES_REQUESTED** [review]

File: assignment2.js (hunk 7 above)
Comment (rishabh1164) [reply assignment2.js]: updated the program for question 1 .

File: Assignment2.js (hunk 2 above)
Comment (rishabh1164) [reply Assignment2.js]: replaced let with const .

**Review by ajay-nx — APPROVED** [review]

**Pull request merged**